homeassistant/components/generic/camera.py

```diff
@@ -65,24 +66,62 @@
 async def async_setup_platform(hass, config, async_add_entities, discovery_info=None):
     """Set up a generic IP Camera."""
 
-    await async_setup_reload_service(hass, DOMAIN, PLATFORMS)
+    _LOGGER.warning(
+        "Loading generic IP camera via configuration.yaml is deprecated, "
+        "it will be automatically imported.  Once you have confirmed correct "
+        "operation, please remove 'generic' (IP camera) section(s) from "
+        "configuration.yaml"
+    )
+    streamsource = config.get(CONF_STREAM_SOURCE)
+    config_new = {
```

Comment:
Why do this? you could just modify the config in place to convert the templates back to a string and pass that to the import

Reply:
Thanks! Done.